serverlessworkflow/modules/ROOT/pages/core/timeouts-support.adoc

```diff
@@ -0,0 +1,359 @@
+= Timeouts on events for {context}
+:compat-mode!:
+// Metadata:
+:description: Using timeouts in {context}
+:keywords: kogito, workflow, serverless, timeout, timer, expiration
+
+When defining a state in a workflow you can set a timeout which is a property used to control the maximum time this state should take to be complete. In case of the time is elapsed and overdue, this state is considered timed-out, and the workflow moves to the next stated defined in the transition. The properties that you can use when defining a state timeout are described in the link:{spec_doc_url}#event-timeout-definition[Serverless Workflow specification].
+
+The event-based states can define a property `eventTimeout` which uses ISO 8601 data and time format to specify a duration of time, for instance, you can if you can represent 15 minutes like `PT15M`,  to represent 15 minutes or 2 days, 3 hours and 4 minutes `P2DT3H4M`.
+
+[NOTE]
+====
+Event timeouts can not be defined as a specific point in time, but it should be an amount of time, a duration, which is considered to start when the referred state becomes active in the workflow.
+====
+
+[IMPORTANT]
+====
+{product_name} currently, supports timeout only for *Callback* and *Switch* states with events, other states are going to support timeouts in the future releases.
+====
+
+=== Callback state timeout
+Callback state can be used when you need to publish an event to trigger an external service and waits an asynchronous response as an event, the callback. Once this event is consumed, the workflow continues the execution to the next state defined in the transition property. See more on xref:eventing/working-with-callbacks.adoc[Callback state in {context}].
+
+Since the callback event halts the execution util the event is consumed, you can define an `eventTimeout` for it, and in case the event does not arrive in the defined duration time, the workflow continues the execution moving to the next state defined in the transition, see the <<callback-state, example>>.
+
+[#callback-state]
+.Example of callback state with timeout
+[source,json]
+----
+{
+ "name": "CallbackState",
+ "type": "callback",
+ "action": {
+   "name": "callbackAction",
+   "functionRef": {
+     "refName": "callbackFunction",
+     "arguments": {
+       "input": "${\"callback-state-timeouts: \" + $WORKFLOW.instanceId + \" has executed the callbackFunction.\"}"
```

Comment:
I think this argument can be an object instead. This way, you avoid adding these scape chars and educate the users.

Reply:
yeah, @wmedvede can you update the example in this way we align with the guide

Reply:
Hi @ricardozanini, I missed this comment, but what do you mean with an object?

Let me explain:
The idea here is that the invoked function is sysout, why? to make things simple, etc., but also during the review process it was also suggested to remove a kind of "more real service" invocation I had, that in the end did the system out.
So, the idea is, that this two SWs are intended to show specifically how to use the timeouts and to provide a quickly way to see understand the code, execute, and see the results in the system output. Let's say a one-to-one matching.

Not sure if I've got your point, but to have an object as argument, I need to create it first, so, or we pass it to the SW as parameter, or we use an inject state, or we create it somehow in another point of the SW, etc.
But with this options, I loose the end-to-end mapping from the SW definition to the system output, and the simplicity of reading the flow.

What do you suggest?

Reply:
Oh ok, this is used in sysout, I missed that. This is fine.